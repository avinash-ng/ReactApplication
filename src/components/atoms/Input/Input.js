import React from 'react'
import PropTypes from 'prop-types'

const Input = (props) => {
    // console.log(props.type + " " + props.value + " " + props.name)
    return ( <
        input type={props.type}
        className={props.className}
        onChange = { props.onChange }
        value = { props.value }
        name = { props.name }
        placeholder = { props.placeholder }
        />
    )
}

Input.propTypes = {
    type: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.string,
    name: PropTypes.string

}
export default Input