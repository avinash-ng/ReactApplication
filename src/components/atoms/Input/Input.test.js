import Input from './Input'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, mount, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Input /> ', () => {
    it('renders Input element', () => {
        const RenderComponent = shallow(
            <Input ></Input>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly and check onchange function', () => {
        const type = "text"
        const onchange = sinon.spy()
        const value = "custom value"
        const name = "input field"
        const RenderComponent = mount(<Input type={type} name={name} value={value} onChange={onchange}></Input>)
        RenderComponent.find('input').simulate('change')
        expect(onchange).toHaveProperty('callCount', 1)

    })


    it('Matches snapshot', () => {
        const type = "text"
        const onChange = jest.fn()
        const value = " custom value"
        const name = "input field"
        const RenderComponent = shallow(<Input type={type} name={name} value={value} onChange={onChange}></Input>)
        expect(RenderComponent).toMatchSnapshot()
    
    })
})