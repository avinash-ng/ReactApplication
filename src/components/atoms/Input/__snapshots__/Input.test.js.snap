// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input />  Matches snapshot 1`] = `ShallowWrapper {}`;
